hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/read/TestECBlockReconstructedInputStream.java

```diff
@@ -0,0 +1,317 @@
+import org.apache.hadoop.hdds.client.ECReplicationConfig;
+import org.apache.hadoop.hdds.protocol.DatanodeDetails;
+import org.apache.hadoop.hdds.protocol.MockDatanodeDetails;
+import org.apache.hadoop.ozone.client.io.ECBlockReconstructedInputStream;
+import org.apache.hadoop.ozone.client.io.ECBlockReconstructedStripeInputStream;
+import org.apache.hadoop.ozone.om.helpers.OmKeyLocationInfo;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.SplittableRandom;
+import java.util.concurrent.ThreadLocalRandom;
+
+import static org.apache.hadoop.ozone.client.rpc.read.ECStreamTestUtil.generateParity;
+
+/**
+ * Test for the ECBlockReconstructedInputStream class.
+ */
+public class TestECBlockReconstructedInputStream {
+
+  private ECReplicationConfig repConfig;
+  private ECStreamTestUtil.TestBlockInputStreamFactory streamFactory;
+  private long randomSeed;
+  private ThreadLocalRandom random = ThreadLocalRandom.current();
+  private SplittableRandom dataGenerator;
+
+  @Before
+  public void setup() throws IOException {
+    repConfig = new ECReplicationConfig(3, 2);
+    streamFactory = new ECStreamTestUtil.TestBlockInputStreamFactory();
+
+    randomSeed = random.nextLong();
+    dataGenerator = new SplittableRandom(randomSeed);
+  }
+
+  private ECBlockReconstructedStripeInputStream createStripeInputStream(
+      Map<DatanodeDetails, Integer> dnMap, long blockLength) {
+    OmKeyLocationInfo keyInfo =
+        ECStreamTestUtil.createKeyInfo(repConfig, blockLength, dnMap);
+    streamFactory.setCurrentPipeline(keyInfo.getPipeline());
+    return new ECBlockReconstructedStripeInputStream(repConfig, keyInfo, true,
+        null, null, streamFactory);
+  }
+
+  @Test
+  public void testBlockLengthReturned() throws IOException {
+    Map<DatanodeDetails, Integer> dnMap = createIndexMap(4, 5);
+    try(ECBlockReconstructedStripeInputStream stripeStream
+        = createStripeInputStream(dnMap, 12345L)) {
+      try (ECBlockReconstructedInputStream stream =
+          new ECBlockReconstructedInputStream(repConfig, stripeStream)) {
+        Assert.assertEquals(12345L, stream.getLength());
+      }
+    }
+  }
+
+  @Test
+  public void testBlockIDReturned() throws IOException {
+    Map<DatanodeDetails, Integer> dnMap = createIndexMap(1, 4, 5);
+    try(ECBlockReconstructedStripeInputStream stripeStream
+            = createStripeInputStream(dnMap, 12345L)) {
+      try (ECBlockReconstructedInputStream stream =
+               new ECBlockReconstructedInputStream(repConfig, stripeStream)) {
+        Assert.assertEquals(new BlockID(1, 1), stream.getBlockID());
+      }
+    }
+  }
+
+  @Test
+  public void testReadDataByteBufferMultipleStripes() throws IOException {
+    int readBufferSize = random.nextInt(4096);
+    // 3 stripes and a partial chunk
```

Comment:
Block len less than ECChunksize test also would be good. I remember that kid of test covered in one of previous patches. If that holds good, then fine.

Reply:
We have test that check the chunk boundaries in the tests for the StripeReader. Also in this class `testReadDataByteBufferUnderBufferSize` has a block of only 1024 bytes which is under the chunksize.

Reply:
Got it. Thanks